### ISSUE TITLE
Add new cookies group

### ISSUE THREAD

**Issue by agracia-foticos** [dargmuesli/nuxt-cookie-control#216]
### Describe the feature

There is two types of cookies:

necessary and optional.

Please allow add new group custom cookies.

In a lot of websites, the cookies are grouped like "necesary", "ads", "analytics", "geo", etc... not only 2 "static" group of cookies

### Additional information

- [ ] Would you be willing to help implement this feature?

### Final checks

- [X] Check existing [discussions](https://github.com/nuxt/framework/discussions) and [issues](https://github.com/nuxt/framework/issues).

**Comment by dargmuesli** [comment]
PRs welcome!

**Comment by agracia-foticos** [comment]
here you are an example of other consent banner cookies with custom categories
https://cookieconsent.orestbida.com/reference/configuration-reference.html#categories

**Comment by agracia-foticos** [comment]
Or maybe extend, not only optional and necessary.
Extend to 
- Necessary
- Functionality
- Analytics
- Marketing

**Comment by agracia-foticos** [comment]
![image](https://github.com/dargmuesli/nuxt-cookie-control/assets/77386602/497ce0f3-c15c-47ff-a51d-f1bfd5c35a00)
Extend in this enum to :

export declare enum CookieType {
    NECESSARY = "necessary",
    FUNCTIONALITY = "functionality",    
ANALYTICS = "analytics",
MARKETING = "marketing"
}

**Comment by dargmuesli** [comment]
I think the categories should be configurable eventually, as configurability is a goal of this project.

**Comment by agracia-foticos** [comment]
> I think the categories should be configurable eventually, as configurability is a goal of this project.

@dargmuesli 
Do you know when you will implement this configurability?

**Comment by dargmuesli** [comment]
Yes, I won't implement this feature myself, that's why I wrote "PR welcome" and tagged this issue with the "help wanted" label.

**Comment by agracia-foticos** [comment]
@dargmuesli 
I create new pull request to fix this issue

**Comment by dargmuesli** [comment]
Closing due to inactivity. Will reopen once there is continued demand for it. I'm also working on a full rewrite at the moment which may include this feature.
